### ISSUE TITLE
Unexpected typecheck error with missing namespace JSX

### ISSUE THREAD

**Issue by ray007** [vega/ts-json-schema-generator#2544]
I rerun my schema generating task again after quite a while and got an unexpected error on program init:
```cmd
npm notice run ts-json-schema-generator -j extended -e all --full-description --path ./src/types.ts --out dist/wc-types.json
error TSJ - 108: Type check error

  C:/devel/projectdir/node_modules/@types/mdx/types.d.ts:23:38
    23 type StringComponent = Extract<keyof JSX.IntrinsicElements, ElementType extends never ? string : ElementType>;
                                            ~~~
    Cannot find namespace 'JSX'.
  C:/devel/projectdir/node_modules/@types/mdx/types.d.ts:28:23
    28 export type Element = JSX.Element;
                             ~~~
    Cannot find namespace 'JSX'.
  C:/devel/projectdir/node_modules/@types/mdx/types.d.ts:47:28
    47     ? new(props: Props) => JSX.ElementClass
                                  ~~~
    Cannot find namespace 'JSX'.
  C:/devel/projectdir/node_modules/@types/mdx/types.d.ts:73:46
    73         [Key in StringComponent]?: Component<JSX.IntrinsicElements[Key]>;
                                                    ~~~
    Cannot find namespace 'JSX'.

Error: Type check error
    at createProgram (C:\Users\MyUser\AppData\Roaming\npm\node_modules\ts-json-schema-generator\dist\factory\program.js:71:19)
    at createGenerator (C:\Users\MyUser\AppData\Roaming\npm\node_modules\ts-json-schema-generator\dist\factory\generator.js:11:72)
    at Object.<anonymous> (C:\Users\MyUser\AppData\Roaming\npm\node_modules\ts-json-schema-generator\dist\ts-json-schema-generator.js:62:55)
    at Module._compile (node:internal/modules/cjs/loader:1854:14)
    at Object..js (node:internal/modules/cjs/loader:1985:10)
    at Module.load (node:internal/modules/cjs/loader:1577:32)
    at Module._load (node:internal/modules/cjs/loader:1379:12)
    at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
    at Module.require (node:internal/modules/cjs/loader:1600:12)
    at require (node:internal/modules/helpers:153:16)
```

But this is a project for web components using the lit-library, no JSX anywhere in use.

The error goes away when adding the cmd line parameter `--no-type-check`, but I'm not sure that's a good idea...

OTOH the project is part of a `npm workspace`, that includes a storybook project, which drags the mdx-types in with a dependency.

So it would seem to me, the schema-generator loads more types than it actually needs.

So I tried pointing the schema generator at `tsconfig.app.json` instead of `tsconfig.json`.
But now the global types aren't found anymore:
```cmd
error TSJ - 108: Type check error

    Cannot find global type 'Array'.
    Cannot find name 'Boolean'. Did you mean 'GLboolean'?
    Cannot find global type 'CallableFunction'.
    Cannot find global type 'Function'.
    Cannot find global type 'IArguments'.
    Cannot find global type 'NewableFunction'.
    Cannot find global type 'Number'.
    Cannot find global type 'Object'.
    Cannot find global type 'RegExp'.
    Cannot find global type 'String'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions
```

Anybody an idea what's suddenly going wrong?

**Comment by domoritz** [comment]
If you need to run `--no-type-check` that usually indicates some kind of type error that typescript would also show you. Or there is a differently configured typescript. I think you may be able to fix this by running ts-json-schema-generator with your tsconfig provided (`-f`).

**Comment by ray007** [comment]
Yes, I did try with `-f tsconfig.app.json`, which got me the missing globals.

Depending on whether or not I have a line
```
  "types": ["node"],
```
in there, the errors also contain
```
    Cannot find type definition file for 'node'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions
```
or not.

I can get things to work again, by reducing ts-config to 1 file `tsconfig.json` (without `.app` and `.node` variants), but that brings back the errors about helper-scripts outside the `src` directory not being found be the project service.
Well, guess that's the lesser problem to live with.

**Comment by domoritz** [comment]
So seems like the bug is not here, right? Okay if we close the issue?

**Comment by ray007** [comment]
I still think there's a problem working with a non-standard TS-config and/or npm workspace, but it's not blocking my work.
